### PR TITLE
KAFKA-4664: Update docs/protocol.html with KIP-97 information

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: docs/protocol.html [hunk 1]
```diff
@@ -109,13 +109,13 @@
...
+<p>In order to provide cross-version compatibility, brokers can handle a range of versions.  For example, a 0.10.0 broker can handle a v0, v1, or v2 FetchRequest.  On the other hand, this broker can only handle a v1 shutdown request, not a v0.  The response message that the broker sends will have the same version as the request that was sent.
 
-<p>Currently all versions are baselined at 0, as we evolve these APIs we will indicate the format for each version individually.</p>
+<p>Starting in 0.10.2, clients can also handle a range of versions.  This was implemented as part of KIP-97.  As the first thing it does, the client sends an ApiVersionsRequest to the broker to learn which versions of each API are supported.  The client will attempt to send a request that is supported by both itself and the broker.
```
Comment (ijuma) [inline docs/protocol.html]: The style of writing was changed from describing intent in a few places to simply stating facts. @gwenshap, what are your thoughts?

Comment (gwenshap) [reply docs/protocol.html]: Agree. I liked starting with "The protocol is designed to..." - explaining what we were trying to do before diving into details of how we do things.

Comment (cmccabe) [reply docs/protocol.html]: OK

**Comment by ijuma** [comment]
Thanks for the PR. I had a go at a few improvements:

https://github.com/ijuma/kafka/commit/bce66c26449c9a5e11ecba900e25e55f6d9416fc

What do you think?

**Comment by cmccabe** [comment]
LGTM (non-binding)

**Comment by ijuma** [comment]
Merged the PR with the changes in https://github.com/apache/kafka/pull/2387#issuecomment-275658635 to trunk and 0.10.2.

**Comment by gwenshap** [comment]
<3 Looks great! Thank you both

**Pull request closed without merging**